gnome-initial-setup/pages/account/gis-account-page.c

```diff
@@ -102,6 +102,32 @@ on_validation_changed (gpointer        page_area,
   update_page_validation (page);
 }
 
+
+static void
+hide_password_switch_if_needed (GisAccountPage *page)
+{
+  GKeyFile *conf_file = gis_driver_get_vendor_conf_file (GIS_PAGE (page)->driver);
+  GisAccountPagePrivate *priv = gis_account_page_get_instance_private (page);
+  g_autoptr(GError) error = NULL;
+
+  /* do nothing if the configuration file doesn't exist */
+  if (conf_file == NULL)
+    return;
+
+  /* check the conf file to see if the password switch should be hidden */
+  if (!g_key_file_get_boolean (conf_file, "page.account", "show-password-switch", &error) &&
```

Comment:
It's a bit odd that the other use we have with this uses a different namespace (`Welcome` for the branding-welcome page if I'm not mistaken.)

Reply:
I saw that but it doesn't invalidate this code, right? I actually think that _Welcome_ as a group that refers to the _branding-welcome_ can be misleading, and we should change it to be _page.branding-welcome_ instead. I can do that in a different PR.

Reply:
It might be worth encapsulating the vendor config keyfile in a class so that all the groups and keys are maintained in one place. Right now it's a little hard to see what's being used where. I did this in the reformatter for the unattended configuration https://github.com/endlessm/eos-installer/blob/master/gnome-image-installer/util/gis-unattended-config.c and think it's all more legible as a result.

I think it's good to name the group after the page id – either `account` or `page.account` seems fine to me. It's unfortunate that the `branding-welcome` page uses `Welcome` but while we can make it check a new group/key to match a new convention, we still need to check the old group/key for backwards-compatibility with old vendor config files.

Reply:
Or, at least, put all the `#define`s for group & key names into one header, which can serve as the definition for what can be put in the keyfile. Right now it's smeared across the code.

Reply:
@wjt I understand your suggestion, but what you did for the installer seems to suggest you had a more complex use-case than just retrieving configuration. I thought that not confining pages to using a "proxy" for the config here was a good thing, at least until we realize all the needs the pages have. I certainly don't want to do a wrapper for all the methods in `GKeyFile` until we understand if those are needed.

Since all pages are in the tree and no keys are being shared at the moment, I think it's okay to let pages worry about their keys (but I will move the group/keys into defines in the page's code here, if that makes it a bit more organized).